internal/service/streamprocessor/resource.go

```diff
@@ -0,0 +1,258 @@
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	streamProcessorReq, diags := NewStreamProcessorReq(ctx, &plan)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	var needsStarting bool
+	if !plan.State.IsNull() && !plan.State.IsUnknown() {
+		switch plan.State.ValueString() {
+		case StartedState:
+			needsStarting = true
+		case CreatedState:
+			needsStarting = false
+		default:
+			resp.Diagnostics.AddError("When creating a stream processor, the only valid states are CREATED and STARTED", "")
+		}
+	}
+
+	connV2 := r.Client.AtlasV2
+	projectID := plan.ProjectID.ValueString()
+	instanceName := plan.InstanceName.ValueString()
+	processorName := plan.ProcessorName.ValueString()
+	_, _, err := connV2.StreamsApi.CreateStreamProcessor(ctx, projectID, instanceName, streamProcessorReq).Execute()
+
+	if err != nil {
+		resp.Diagnostics.AddError("error creating resource", err.Error())
+		return
+	}
+
+	streamProcessorParams := &admin.GetStreamProcessorApiParams{
+		GroupId:       projectID,
+		TenantName:    instanceName,
+		ProcessorName: processorName,
+	}
+
+	streamProcessorResp, err := WaitStateTransition(ctx, streamProcessorParams, connV2.StreamsApi, []string{InitiatingState, CreatingState}, []string{CreatedState, FailedState})
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating stream processor", err.Error())
+	}
+
+	if needsStarting {
+		_, _, err := connV2.StreamsApi.StartStreamProcessorWithParams(ctx,
+			&admin.StartStreamProcessorApiParams{
+				GroupId:       projectID,
+				TenantName:    instanceName,
+				ProcessorName: processorName,
+			},
+		).Execute()
+		if err != nil {
+			resp.Diagnostics.AddError("Error starting stream processor", err.Error())
+		}
+		streamProcessorResp, err = WaitStateTransition(ctx, streamProcessorParams, connV2.StreamsApi, []string{CreatedState}, []string{StartedState})
+		if err != nil {
+			resp.Diagnostics.AddError("Error changing state of stream processor", err.Error())
+		}
+	}
+
+	newStreamProcessorModel, diags := NewStreamProcessorWithStats(ctx, projectID, instanceName, streamProcessorResp, plan.Options)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, newStreamProcessorModel)...)
+}
+
+func (r *streamProcessorRS) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state TFStreamProcessorRSModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	connV2 := r.Client.AtlasV2
+
+	projectID := state.ProjectID.ValueString()
+	instanceName := state.InstanceName.ValueString()
+	streamProcessor, apiResp, err := connV2.StreamsApi.GetStreamProcessor(ctx, projectID, instanceName, state.ProcessorName.ValueString()).Execute()
+	if err != nil {
+		if apiResp != nil && apiResp.StatusCode == http.StatusNotFound {
+			resp.State.RemoveResource(ctx)
+			resp.Diagnostics.AddError("resource not found", err.Error())
```

Comment:
This line should be removed. (sorry for earlier confusion)

Reply:
Removed in [1e09f4e](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2501/commits/1e09f4e3d4f54ad618499d2a7a71cb8c1237533c)